File: module/iOSThemeClub/src/main/java/com/amz/ios/themeclub/base/BaseFragment.java
package com.amz.ios.themeclub.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.amz.ios.ioslite.common.debug.DebugLog;
import com.amz.ios.themeclub.R;

/**
 * Created by server on 16-11-17.
 */


public abstract class BaseFragment extends Fragment {
    public View loadingView;
    private boolean mCreatViewFinish;
    private final String TAG = getClass().getSimpleName();
    private boolean mIsVisibleToUser;

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        mIsVisibleToUser = isVisibleToUser;
        lazyLoad();
    }


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = createView(inflater,container,savedInstanceState);
        loadingView = inflater.inflate(R.layout.loading_view,null);
        init(v);
        return v;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mCreatViewFinish = true;
        lazyLoad();
    }

    private void lazyLoad() {
        DebugLog.w(TAG,"=================lazyLoad:"+mCreatViewFinish+"/"+mIsVisibleToUser);
        if (mCreatViewFinish&&mIsVisibleToUser){
            fragmentLoadData();
        }
    }

    protected abstract void fragmentLoadData();

    protected abstract View createView(LayoutInflater inflater,@Nullable ViewGroup container,@Nullable Bundle saveInstaceState);

    protected abstract void init(View v);

}

File: app/src/main/java/com/oslauncher/applauncher/themelauncher/Base/BaseFragment.kt
package com.oslauncher.applauncher.themelauncher.Base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewbinding.ViewBinding
import com.oslauncher.applauncher.themelauncher.tool.languageTool.LanguageUtil

abstract class BaseFragment<b : ViewBinding> : Fragment() {
    lateinit var binding: b

    abstract fun setViewBinding(inflater: LayoutInflater, container: ViewGroup?): b

    //setupView here
    abstract fun initView()

    //listen to user action here
    abstract fun viewListener()

    //listen to database change here
    abstract fun dataObservable()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        LanguageUtil.setLocale(requireActivity())
        binding = setViewBinding(inflater, container)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        viewListener()
        dataObservable()
    }
}